nats-experimental/src/options.rs

```diff
@@ -67,24 +67,79 @@ impl Default for Options {
 }
 
 impl Options {
+    /// Enables customization of NATS connection.
+    ///
+    /// # Example
```

Comment:
Nit: heading should preferably be `# Examples` to be consistent with std and other core things.

Reply:
It was  literally last second. I was just clicking Rebase button 😄